package.json

```diff
@@ -529,7 +529,8 @@
       {
         "command": "mssql.createAzureFunction",
         "title": "%mssql.createAzureFunction%",
-        "category": "MS SQL"
+        "category": "MS SQL",
+        "when": "view == objectExplorer && viewItem == database"
```

Comment:
Why database? Don't you want this on tables instead so it has the correct table context to insert?

(also - side note. Is there a reason we're not supporting views for this?)